looper/looper.py

```diff
@@ -555,20 +566,700 @@ def __call__(self, args, remaining_args):
+                                    self.prj.metadata.results_subdir,
+                                    sample_name, "stats.tsv")
+                    if os.path.isfile(stats_file):
+                        t = _pd.read_table(stats_file, header=None,
+                                           names=['key', 'value', 'pl'])
+                        t.drop_duplicates(subset=['key', 'pl'],
+                                          keep='last', inplace=True)
+                        try:
+                            time = str(t[t['key'] == 'Time'].iloc[0]['value'])
+                            html_file.write(STATUS_ROW_VALUE.format(
+                                            row_class="",
+                                            value=str(time)))
+                        except IndexError:
+                            warnings.append("The stats_summary.tsv file is incomplete")                        
+                    else:
+                        html_file.write(STATUS_ROW_VALUE.format(
+                                            row_class=button_class,
+                                            value="Unknown"))
+                    html_file.write(STATUS_ROW_FOOTER)
+
+                html_file.write(STATUS_FOOTER)
+                html_file.write(HTML_FOOTER)
+                html_file.close()
+                if warnings:
+                    _LOGGER.warn("Warning: " + ''.join(list(set(warnings))))
+
+        def create_sample_html(all_samples, sample_name, sample_stats,
+                               index_html):
+            # Produce an HTML page containing all of a sample's objects
+            # and the sample summary statistics
+            reports_dir = os.path.join(self.prj.metadata.output_dir, "reports")
+            html_filename = sample_name + ".html"
+            html_page = os.path.join(
+                reports_dir, html_filename.replace(' ', '_').lower())
+            sample_page_relpath = os.path.relpath(
+                html_page, self.prj.metadata.output_dir)
+            single_sample = all_samples[all_samples['sample_name'] == sample_name]
+            if not os.path.exists(os.path.dirname(html_page)):
+                os.makedirs(os.path.dirname(html_page))
+            with open(html_page, 'w') as html_file:
+                html_file.write(HTML_HEAD_OPEN)
+                html_file.write("\t\t<style>\n")
+                html_file.write(SAMPLE_TABLE_STYLE)
+                html_file.write("\t\t</style>\n")
+                html_file.write(create_navbar(all_samples, reports_dir))
+                html_file.write(HTML_HEAD_CLOSE)
+                html_file.write("\t\t<h4>{}</h4>\n".format(str(sample_name)))
+                if single_sample.empty:
+                    # When there is no objects.tsv file, search for the
+                    # presence of log, profile, and command files
+                    log_name = os.path.basename(str(glob.glob(os.path.join(
```

Comment:
This patterns suggests that we could provide in `peppy` a function on `Project` to get file(s) by extension.

Reply:
can we just use `prj.logfile` in this case? I'm pretty sure the log filename is an attribute of the project.

Reply:
I may be simply missing it, but I don't see/can't access a logfile attribute?  Can you confirm how you reference the sample logfile, if it is possible?

Reply:
you're right. I was thinking of pypiper